examples/bridge-app/linux/Device.cpp

```diff
@@ -110,6 +110,25 @@ void DeviceOnOff::SetOnOff(bool aOn)
     }
 }
 
+void DeviceOnOff::MQTTPublisher(mqtt::async_client& client, const std::string& topic, const std::string& payload) {
+    try {
+        ChipLogProgress(DeviceLayer, "[MQTT] Connecting...");
+        client.connect()->wait();
+        ChipLogProgress(DeviceLayer, "[MQTT] Connected.");
+
+        ChipLogProgress(DeviceLayer, "[MQTT] Publishing message...");
+        client.publish(mqtt::make_message(topic, payload))->wait();
+        ChipLogProgress(DeviceLayer, "[MQTT] Message published.");
+
+        ChipLogProgress(DeviceLayer, "[MQTT] Disconnecting...");
+
+        client.disconnect()->wait();
+        ChipLogProgress(DeviceLayer, "[MQTT] Disconnected.");
```

Comment:
Connection and disconnection should happen during init and shutdown. The client should stay connected during the runtime of the application.

Reply:
Shutdown() has been defined here: https://github.com/project-chip/connectedhomeip/blob/master/src/include/platform/PlatformManager.h#L443-L447

Reply:
Does the event loop exists the application on stop signal or does it return to the caller (main function)? If the latter, then the disconnection could happen after the even loop returns.

Otherwise, leave it for future work as it is not so critical. MQTT broker automatically detects and disconnects dead clients when the TTL message does not arrive in time.

Reply:
The event loop exit the application on stop signal: https://github.com/MonicaisHer/connectedhomeip/pull/2/files#r1291086217